Feature/reddit

jessicafarias opened the pull request:
We created 3 models which created User, Post and Comment classes inheriting from ApplicationRecord and then generating the ActiveRecord:: Migration[6.0] for each class.
We then added ActiveRecord validations :
- to verify the presence of the users in each post and comment
- to verify the presence of the email and username

aracelicaes committed:
Created new rails app for Micro_reddit
Installed Gems
Created app

aracelicaes committed:
Changed Database from SQLITE3 TO POSTGRESQL

jessicafarias committed:
Create User model

Create User table
Adding validation helpers tu User parameters
Create some users and uploaded to database

aracelicaes committed:
Create Post Model
Generated Post Model
Create Posts table: title, body, user_id
Added Post validations to ensure there's a user assigned to each post. Body length doesn't exceed 200 characters.
Added Associations Between User and Post.
:

jessicafarias committed:
Create a new comment model

Adding validations to comments
Adding has_many comments into model post and user
Create a new squema using db:migration

aracelicaes committed:
Adding Linters
Added Rubocop
Added Linters

aracelicaes committed:
Edit ReadME
Created ReadME content.

Dandush03 commented on app/models/comment.rb:
```diff
@@ -0,0 +1,6 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  validates :user, presence: true
+  validates :post, presence: true
```
#### Missing Validation
- [x] `body` - Should not be able to receive an empty value. Missing Validation

Dandush03 commented on app/models/post.rb:
```diff
@@ -0,0 +1,6 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  validates :body, length: { maximum: 200 }
+  validates :user, presence: true
```
#### Missing Validation
- [x] `title` - Should not be able to receive an empty value. Missing Validation
- [x] `body` - Should not be able to receive an empty value. Missing Validation

Dandush03 reviewed (changes requested):
# Status: :red_circle:  Require Changes :red_circle: 
## Great Work so Far
### There are some validation that are missing, please check my comment below
# Happy Coding :rocket: :computer:

jessicafarias committed:
Adding validations

Adding presence validation for title and body in post class
Adding presence validation for the body in a comment

uryelah reviewed (approved):
Hi **Jessica Farías Rosado & Ara Camarillo Estudillo**, I'm Sarah your reviewer for this PR 👩‍🚀 

You did an awesome job with the validations, specially the one for the emal!

Go forth and rock your next project!  👩‍🎤  🎸  

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)
